batch-processing/src/test/java/org/wildfly/quickstarts/batch-processing/BasicRuntimeIT.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.quickstarts.helloworld;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * The very basic runtime integration testing.
+ * @author emartins
+ */
+public class BasicRuntimeIT {
+
+    private static final String DEFAULT_SERVER_HOST = "http://localhost:8080/batch-processing";
```

Comment:
@emmartins After reading the document  `WildFly Quickstarts Enhancements CY2023Q3` I followed the instruction to build the project and do the test with server provisioning:

```bash
$ mvn clean package -Pprovisioned-server
```

```bash
$ mvn wildfly:start -DjbossHome=target/server
```

```bash
$  mvn verify -Pintegration-testing -Dserver.host=http://localhost:8080
```

The test passed, however with packet capture I see the server response is actually the JBoss Homepage:

<img width="1735" alt="image" src="https://github.com/wildfly/quickstart/assets/201907/5ba939af-e953-46a3-96ce-c4dc73caa87e">

Is this what the test is expected?

Reply:
In addition it seems there is error during the sample deployment process:

```bash
03:30:27,508 INFO  [org.jboss.as.jpa] (ServerService Thread Pool -- 62) WFLYJPA0010: Starting Persistence Unit (phase 1 of 2) Service 'ROOT.war#primary'
03:30:27,526 INFO  [org.hibernate.jpa.internal.util.LogHelper] (ServerService Thread Pool -- 62) HHH000204: Processing PersistenceUnitInfo [name: primary]
03:30:27,601 INFO  [org.hibernate.Version] (ServerService Thread Pool -- 62) HHH000412: Hibernate ORM core version 6.3.1.Final
03:30:27,915 INFO  [org.hibernate.cache.internal.RegionFactoryInitiator] (ServerService Thread Pool -- 62) HHH000025: Second-level cache region factory [org.infinispan.hibernate.cache.v62.InfinispanRegionFactory]
03:30:27,998 INFO  [org.hibernate.envers.boot.internal.EnversServiceImpl] (ServerService Thread Pool -- 62) Envers integration enabled? : true
03:30:28,197 INFO  [org.jboss.as.jpa] (ServerService Thread Pool -- 62) WFLYJPA0010: Starting Persistence Unit (phase 2 of 2) Service 'ROOT.war#primary'
03:30:29,457 INFO  [org.jboss.weld.Bootstrap] (Weld Thread Pool -- 10) WELD-000119: Not generating any bean definitions from org.jboss.as.quickstarts.batch.controller.BatchController because of underlying class loading error: Type jakarta.faces.context.FacesContext from [Module "deployment.ROOT.war" from Service Module Loader] not found.  If this is unexpected, enable DEBUG logging to see the full error.
03:30:29,817 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 65) WFLYUT0021: Registered web context: '/' for server 'default-server'
03:30:29,877 INFO  [org.jboss.as.server] (ServerService Thread Pool -- 37) WFLYSRV0010: Deployed "ROOT.war" (runtime-name : "ROOT.war")
```

I guess it needs some configuration for the above error to be fixed. @chengfang Do you have any suggestion on this?

Reply:
from what i can see, these are just harmless info logs.

Reply:
> Is this what the test is expected?

the system property `-Dserver.host` used by the test is supposed to also include the app context path. Since its value in your command does not have context path, WildFly home page is returned.

The default value in the test should suffice so we can omit this sysprop in the command.

Reply:
> ```shell
> 03:30:29,457 INFO  [org.jboss.weld.Bootstrap] (Weld Thread Pool -- 10) WELD-000119: Not generating any bean definitions from org.jboss.as.quickstarts.batch.controller.BatchController because of underlying class loading error: Type jakarta.faces.context.FacesContext from [Module "deployment.ROOT.war" from Service Module Loader] not found.  If this is unexpected, enable DEBUG logging to see the full error.
> ```

There is one line the seems have error:

```
03:30:29,457 INFO  [org.jboss.weld.Bootstrap] (Weld Thread Pool -- 10) WELD-000119: Not generating any bean definitions from org.jboss.as.quickstarts.batch.controller.BatchController because of underlying class loading error: Type jakarta.faces.context.FacesContext from [Module "deployment.ROOT.war" from Service Module Loader] not found.  If this is unexpected, enable DEBUG logging to see the full error.
```

Reply:
I will investigate, thanks for the warning

Reply:
It was missing the jsf galleon layer.